service/history/replicationTaskExecutor.go

```diff
@@ -153,17 +155,23 @@ func (e *replicationTaskExecutorImpl) handleActivityTask(
 		stopwatch := e.metricsClient.StartTimer(metrics.HistoryRereplicationByActivityReplicationScope, metrics.ClientLatency)
 		defer stopwatch.Stop()
 
-		if resendErr := e.nDCHistoryResender.SendSingleWorkflowHistory(
+		resendErr := e.nDCHistoryResender.SendSingleWorkflowHistory(
 			namespace.ID(retryErr.NamespaceId),
 			retryErr.WorkflowId,
 			retryErr.RunId,
 			retryErr.StartEventId,
 			retryErr.StartEventVersion,
 			retryErr.EndEventId,
 			retryErr.EndEventVersion,
-		); resendErr != nil {
-			e.logger.Error("error resend history for sync activity", tag.Error(resendErr))
-			// should return the replication error, not the resending error
+		)
+		switch resendErr.(type) {
+		case *serviceerror.NotFound:
+			// TODO" cleanup history
```

Comment:
should not worry about history cleanup, it should be handled by the delete history timer task already, or the history GC should handle it.

Reply:
Synced offline. This cannot handle by GC. We need to delete the workflow. I will separate it to a new PR